lib/http_proxy_override.dart

```diff
@@ -14,22 +14,69 @@ Future<String?> _getProxyPort() async {
 }
 
 class HttpProxyOverride extends HttpOverrides {
-  late final String? host;
-  late final String? port;
+  /// The host part of the proxy address.
+  final String? host;
 
-  HttpProxyOverride._(this.host, this.port);
+  /// The port part of the proxy address.
+  final String? port;
 
-  static Future<HttpProxyOverride> createHttpProxy() async {
```

Comment:
I think you should mark this function as `@deprecated` instead remove it directly, to avoid break change at this time